contracts/leaser/src/migrate.rs

```diff
@@ -88,15 +119,66 @@ mod test {
         let addr2 = Addr::unchecked("22222");
         let leases = vec![addr1.clone(), addr2.clone()];
 
-        let mut exp = Batch::default();
-        exp.schedule_migrate_wasm_no_reply(&addr1, MigrateMsg {}, new_code)
-            .unwrap();
-        exp.schedule_migrate_wasm_no_reply(&addr2, MigrateMsg {}, new_code)
-            .unwrap();
+        let exp = add_expected(MigrationResult::default(), &addr1, new_code);
+        let mut exp = add_expected(exp, &addr2, new_code);
+        exp.last_instance = Some(addr2);
 
         assert_eq!(
             Ok(exp),
-            super::migrate_leases(leases.into_iter().map(Ok), new_code)
+            super::migrate_leases(leases.into_iter().map(Ok), new_code, 2)
         );
     }
+
+    #[test]
+    fn paging() {
+        let new_code = 242;
+        let addr1 = Addr::unchecked("11111");
+        let addr2 = Addr::unchecked("22222");
+        let addr3 = Addr::unchecked("333333333");
+        let addr4 = Addr::unchecked("4");
+        let addr5 = Addr::unchecked("5555");
+        let addr6 = Addr::unchecked("6");
+        let addr7 = Addr::unchecked("777");
+        let leases: Vec<Addr> = [&addr1, &addr2, &addr3, &addr4, &addr5, &addr6, &addr7]
+            .map(Clone::clone)
+            .into();
+
+        {
+            let exp = add_expected(MigrationResult::default(), &addr1, new_code);
+            let mut exp = add_expected(exp, &addr2, new_code);
+            exp.last_instance = Some(addr2);
+            assert_eq!(
+                Ok(exp),
+                super::migrate_leases(leases.clone().into_iter().map(Ok), new_code, 2)
+            );
+        }
+
+        {
+            let exp = add_expected(MigrationResult::default(), &addr3, new_code);
+            let exp = add_expected(exp, &addr4, new_code);
+            let mut exp = add_expected(exp, &addr5, new_code);
+            exp.last_instance = Some(addr5);
+            assert_eq!(
+                Ok(exp),
+                super::migrate_leases(leases.clone().into_iter().skip(2).map(Ok), new_code, 3)
+            );
```

Comment:
Expected expressions are second arguments to the macro.
```suggestion
            assert_eq!(
                super::migrate_leases(leases.clone().into_iter().skip(2).map(Ok), new_code, 3),
                Ok(exp)
            );
```

Reply:
same